packages/shared-lib/src/lib/JobQueueWithClasses.ts

```diff
@@ -102,6 +117,18 @@ export class JobQueueWithClasses {
 			console.error(error)
 		})
 	}
+	/** Returns the count of waiting jobs in the queue (ie not started yet) */
+	getWaiting(): number {
+		return this.#queue.length
+	}
+	/** Returns a Promise that resolves when the queue is eventually empty */
+	async waitForDone(): Promise<void> {
+		if (!this.#queue.length) return Promise.resolve()
```

Comment:
Is this correct? If there is a job currently executing, it looks like this will resolve before it finishes.